metricflow/engine/metricflow_engine.py

```diff
@@ -5,11 +5,13 @@
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
 from enum import Enum
-from typing import List, Optional, Sequence
+from typing import List, Optional, Sequence, Tuple
```

Comment:
<commit level comment> - it would be nice to have a commit message that provides details on why all of the added complexity is necessary to support jamming a dimension that exists for every metric into the list dimensions call.

I get it because I know the codebase, and because I'm reading the entire change in context, but if someone looks at a subset of this logic and says "that's weird, why is that there" then ideally a blame view would point to a broader commit with a message that explains what's goin gon.

And when I say "someone" I mean "future Tom and future Paul" because I don't know about you, but I'm going to forget all of this context by next week.

Reply:
Updated the message.